### PR TITLE
RL

### PR DESCRIPTION
**Pull request by SergioMartin86**
- Integrated Gaussian Processes module (from George's GPnew branch)
- Added check for ctrl+c on Korali to prevent long experiments to stall terminate requests

### CODE REVIEW

File: source/modules/conduit/conduit._cpp [hunk 1]
```diff
@@ -114,6 +114,9 @@ void korali::Conduit::wait(korali::Sample &sample)
 
   while (sample._state == SampleState::waiting || sample._state == SampleState::initialized)
   {
+    // Check for error signals from python
+    if (korali::isPythonActive && PyErr_CheckSignals() != 0) korali::Logger::logError("User requested break.\n");
+
```
Comment (ikicic) [inline source/modules/conduit/conduit._cpp:119]: The signal should be abstracted away. Conduit (backend) should not be aware of the existence of of Python (frontend).

File: source/modules/solver/gaussianProcess/README.rst [hunk 2]
```diff
@@ -0,0 +1,3 @@
+**************************
+Gaussian Process
+**************************
```
Comment (wadaniel) [inline source/modules/solver/gaussianProcess/README.rst:3]: missing documentation

Comment (SergioMartin86) [reply source/modules/solver/gaussianProcess/README.rst:3]: Added small description. George needs to revisit this.

File: source/engine.hpp [hunk 3]
```diff
@@ -192,6 +192,11 @@ class Engine : public korali::Module
   static Engine *deserialize(knlohmann::json &js);
 };
 
+/**
+* @brief Flag indicating that Korali has been called from Korali
```
Comment (amlucas) [inline source/engine.hpp:196]: from python?

**Pull request merged**